lsp-mode.el

```diff
@@ -2904,10 +2904,11 @@ MODE determines when the callback will be called depending on the
 condition of the original buffer. METHOD is the invoked method.
 If NO-MERGE is non-nil, don't merge the results but return alist workspace->result.
 ID is the request id. "
-  (let (results)
+  (let (results errors)
     (lambda (result)
-      (push (cons lsp--cur-workspace result) results)
-      (when (eq (length results) (length workspaces))
+      (push (cons lsp--cur-workspace result)
+            (if result results errors))
+      (when (eq (+ (length errors) (length results)) (length workspaces))
```

Comment:
here we should check if the server has returned only errors and in that case, we should not call the success callback